sunpy/map/mapbase.py

```diff
@@ -2739,6 +2740,14 @@ def reproject_to(self, target_wcs, *, algorithm='interpolation', return_footprin
         outmap = GenericMap(output_array, target_wcs.to_header(),
                             plot_settings=self.plot_settings)
 
+        # Check rsun mismatch
+        rsun_map = self.meta["rsun_ref"]
+        rsun_target = outmap.meta["rsun_ref"]
+        rsun_diff = abs(rsun_map - rsun_target)
+        if rsun_diff > 200000:
+            warnings.warn("Warning: rsun mismatch greater than 200000 m detected. "
```

Comment:
We try to avoid using warnings.warn, can you replace it with the function warn_user instead?

Reply:
Will do